packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:camera_android_camerax/camera_android_camerax.dart';
+import 'package:camera_android_camerax/src/camera_info.dart';
+import 'package:camera_android_camerax/src/camera_selector.dart';
+import 'package:camera_android_camerax/src/process_camera_provider.dart';
+import 'package:camera_platform_interface/camera_platform_interface.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'android_camera_camerax_test.mocks.dart';
+
+@GenerateMocks(<Type>[
+  ProcessCameraProvider,
+  CameraSelector,
+  CameraInfo,
+])
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  test('Should fetch CameraDescription instances for available cameras',
+      () async {
+    // Arrange
+    final List<dynamic> returnData = <dynamic>[
+      <String, dynamic>{
+        'name': 'Camera 0',
+        'lensFacing': 'back',
+        'sensorOrientation': 0
+      },
+      <String, dynamic>{
+        'name': 'Camera 1',
+        'lensFacing': 'front',
+        'sensorOrientation': 90
+      }
+    ];
+
+    //Create mocks to use
+    final MockProcessCameraProvider mockProcessCameraProvider =
+        MockProcessCameraProvider();
+    final MockCameraSelector mockBackCameraSelector = MockCameraSelector();
+    final MockCameraSelector mockFrontCameraSelector = MockCameraSelector();
+    final MockCameraInfo mockFrontCameraInfo = MockCameraInfo();
+    final MockCameraInfo mockBackCameraInfo = MockCameraInfo();
+    AndroidCameraCameraX.registerWith();
+
+    //Set class level ProcessCameraProvider and camera selectors to created mocks
+    final AndroidCameraCameraX androidCameraCamerax = AndroidCameraCameraX();
+    androidCameraCamerax.setDefaultBackCameraSelector(mockBackCameraSelector);
```

Comment:
The changes look good but I think you need to update the test! This file will also need the license at the top:

```
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
```

Reply:
Sorry about that! Fixed the tests, and added the license.

I also moved the null check back to the method and out of the process camera provider because

1. In my mind (as named) a setter should set the variable even if it is non-null, and
2. If the null check is in the setter, then the tests will all need to mock the host platform reply to getting the instance of ProcessCameraProvider, because we follow that code path when we pass it as an argument to the setter (even though we don't set, because in the setter we do the null check and find the _processCameraProvider already set to the mock.

Open to moving it back if you prefer it that way though!

Reply:
I'm not tied to my approach. This should be fine!